app/src/main/java/de/danoeh/antennapod/activity/WidgetConfigActivity.java

```diff
@@ -0,0 +1,88 @@
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+
+public class WidgetConfigActivity extends AppCompatActivity {
+
+    private static final String WIDGET_PREFS = "widget_preferences";
+    private static final String WIDGET_COLOR = "widget_color";
+
+    private int appWidgetId = AppWidgetManager.INVALID_APPWIDGET_ID;
+
+    private SeekBar opacitySeekBar;
+    private TextView opacityTextView;
+    private RelativeLayout widgetPreview;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        setTheme(UserPreferences.getTheme());
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_widget_config);
+
+        Intent configIntent = getIntent();
+        Bundle extras = configIntent.getExtras();
+        if (extras != null) {
+            appWidgetId = extras.getInt(AppWidgetManager.EXTRA_APPWIDGET_ID,
+                    AppWidgetManager.INVALID_APPWIDGET_ID);
+        }
+
+        Intent resultValue = new Intent();
+        resultValue.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID,appWidgetId);
+        setResult(RESULT_CANCELED,resultValue);
+        if (appWidgetId == AppWidgetManager.INVALID_APPWIDGET_ID) {
+            finish();
+        }
+
+        opacityTextView = findViewById(R.id.widget_opacity_textView);
+        opacitySeekBar = findViewById(R.id.widget_opacity_seekBar);
+        widgetPreview = findViewById(R.id.widgetLayout);
+        opacitySeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+            @Override
+            public void onProgressChanged(SeekBar seekBar, int i, boolean b) {
+                opacityTextView.setText(seekBar.getProgress() + "%");
+                widgetPreview.setBackgroundColor(getAlphaFromPercentage(opacitySeekBar.getProgress()));
+            }
+            @Override
+            public void onStartTrackingTouch(SeekBar seekBar) {
+            }
+            @Override
+            public void onStopTrackingTouch(SeekBar seekBar) {
+            }
+        });
+    }
+
+    public void confirmCreateWidget(View v){
+        AppWidgetManager appWidgetManager = AppWidgetManager.getInstance(this);
+
+        RemoteViews views = new RemoteViews(this.getPackageName(), R.layout.player_widget);
+        views.setInt(R.id.widgetLayout,"setBackgroundColor", getAlphaFromPercentage(opacitySeekBar.getProgress()));
+        appWidgetManager.updateAppWidget(appWidgetId,views);
+
+        SharedPreferences prefs = getSharedPreferences(WIDGET_PREFS, MODE_PRIVATE);
+        SharedPreferences.Editor editor = prefs.edit();
+        editor.putInt(WIDGET_COLOR + appWidgetId,getAlphaFromPercentage(opacitySeekBar.getProgress()));
+        editor.apply();
+
+        Intent resultValue = new Intent();
+        resultValue.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID,appWidgetId);
+        setResult(RESULT_OK,resultValue);
+        finish();
+    }
+
+    public int getAlphaFromPercentage(int opacityPercentage){
```

Comment:
Please add a space between ) and {

I currently try to enforce the Google code style guidelines but there are only a few automatic checks yet because big parts of the app would fail and I don't want to have a commit that touches every file.

Reply:
I will have a look at mentioned guidelines and reformat my code.

Reply:
If you use Android Studio or IntelliJ, you can enable to show violations of the Google checks in the editor. You can ignore checks about the import order but if you could implement the others in your new code sections, that would be great.